e2e/src/bin/historic_trades.rs

```diff
@@ -0,0 +1,276 @@
+use anyhow::Result;
+use core::{
+    contracts::stablex_contract::batch_exchange::event_data::Trade, history::Settlement,
+    models::BatchId,
+};
+use e2e::cmd;
+use pricegraph::{Element, Pricegraph, TokenPair, U256};
+use std::{fs::File, io::Write, path::PathBuf};
+use structopt::StructOpt;
+
+const FULLY_FILLED_THREASHOLD: f64 = 0.95;
+const MIN_AMOUNT: u128 = pricegraph::MIN_AMOUNT as _;
+
+/// Common options for analyzing historic batch data.
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "historic_trades",
+    about = "Utility for comparing historic trades to price estimates.",
+    rename_all = "kebab"
+)]
+struct Options {
+    /// The events registry file store containing past exchange events.
+    #[structopt(long, env = "ORDERBOOK_FILE", parse(from_os_str))]
+    orderbook_file: PathBuf,
+
+    /// The output directory for the computed results.
+    #[structopt(long, env = "OUTPUT_DIR", default_value = "target", parse(from_os_str))]
+    output_dir: PathBuf,
+}
+
+fn main() -> Result<()> {
```

Comment:
Maybe document what this binary does in a readme or as some form of help string or module comment.